WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/tagsfeed/ReaderTagsFeedViewModel.kt

```diff
@@ -190,8 +196,108 @@ class ReaderTagsFeedViewModel @Inject constructor(
         }
     }
 
-    private fun onPostLikeClick() {
-        // TODO
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun onPostLikeClick(postItem: TagsFeedPostItem) {
+        // Immediately update the UI and disable the like button. If the request fails, show error and revert UI state.
+        // If the request fails or succeeds, the like button is enabled again.
+        val isPostLikedUpdated = !postItem.isPostLiked
+        updatePostItemUI(
+            postItemToUpdate = postItem,
+            isPostLikedUpdated = isPostLikedUpdated,
+            isLikeButtonEnabled = false,
+        )
+
+        // After updating the like button UI to the intended state and disabling the like button, send a request to the
+        // like endpoint by using the PostLikeUseCase
+        likePostRemote(postItem, isPostLikedUpdated)
+    }
+
+    private fun updatePostItemUI(
```

Comment:
Tested and works as expected! I wonder if there is a way to reduce the complexity of the  updatePostItemUI method. It seems to me that it handles many responsibilities -> finding the post item, updating it, and setting the new state. Should we break into into smaller functions? For example, a method to just update the post state in the list, and another to handle the UI state update.

Reply:
Thanks for the review. I agree, it ended up being too big. I'm going to break it into smaller methods.

Reply:
@daniloercoli  done 👍 

Please let me know if it looks better.

Reply:
Thanks for addressing my feedback. It looks better now.